deploy-tool/cert/k8_storage.go

```diff
@@ -0,0 +1,151 @@
+package cert
+
+import (
+	"context"
+	"fmt"
+	"github.com/caddyserver/certmagic"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"regexp"
+	"strings"
+)
+
+type SecretStorage struct {
+	Namespace  string
+	KubeClient *kubernetes.Clientset
+}
+
+var matchLabels = map[string]string{
+	"manager": "magiccert",
+}
+var labelSelector = "manager=magiccert"
+
+var specialChars = regexp.MustCompile("[^a-zA-Z0-9_.-]+")
+
+func cleanKey(key string) string {
+	return fmt.Sprintf("cm.k8s.%v", specialChars.ReplaceAllString(key, ""))
```

Comment:
Where does the `key` value come from? Do we have any control over it? I'm a little concerned about being clever and trying to sanitize keys because there's a risk of unexpected collisions, though in this case we only ever expect one ACME client credential per Kubernetes cluster so it's a small risk.

Reply:
I don't think the long `letsencrypt-acme-client-credential` is a good idea here considering the keys can only be 63 characters long. Do you think we can make that a bit smaller?

The key is essentially a file path from my understanding. It's essentially created by certmagic and in our storage layer we're mapping these file paths to secrets in k8s.

Reply:
Oh, I didn't know there was a length restriction here. I still think `k8s` is redundant, all the moreso if you're trying to make the best use of a limited length. Can we at least put a label or annotation (whichever is more appropriate) with a more descriptive name?

Reply:
What I'm reading in k8s docs is that secret names must be "DNS subdomains", which can be "no more than 253" characters, not 63, so it seems like we have enough breathing room to be explicit here.

https://kubernetes.io/docs/concepts/configuration/secret/#overview-of-secrets
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

Reply:
Unfortunately these keys aren't really DNS subdomains, just random strings that we're then going to have to map to DNS subdomains if we need to

So we have the limit defined here https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

I agree though I'll get rid of the `k8s`. These are going to be invisible for us as the users though as it's really only ever going to be modified by the key rotation program